src/tasks/tyeRunTaskProvider.ts

```diff
@@ -97,9 +97,15 @@ export default class TyeRunCommandTaskProvider extends CommandTaskProvider {
                                         waitForProcessClose: false
                                     };
                                 },
-                                onStarted:
-                                    pid => {
-                                        tyePid = pid;
+                                onStdOut: stdOut => {
+                                        let matchingLogLine = null;
+                                        const tyeAppStartedRegex = new RegExp('started successfully with Pid: [0-9]+');
```

Comment:
You could simplify by using capturing groups, e.g. `'started successfully with Pid: (?<pid>[0-9]+)'` and then a single `match()` or `exec()`, rather than use `test()` and then having to separately extract the value from the matched string.

Reply:
I wasn't aware of regex groups. Such a powerful feature!